server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4981,6 +4982,8 @@ private boolean localDirectoryContains(Directory localDirectory, String file, lo
             logger.debug("File {} does not exist in local FS, downloading from remote store", file);
         } catch (IOException e) {
             logger.warn("Exception while reading checksum of file: {}, this can happen if file is corrupted", file);
+            // For any other exception on reading checksum, we delete the file to re-download again
+            localDirectory.deleteFile(file);
```

Comment:
can this happen while the shard is not in recovery?  Wondering if we should be failing the shard in that case similar to when the cksum check fails.

Reply:
This anyway fails recovery as we saw during testing.

Reply:
When control will come to this block, this means the file is non-readable (we are not able to read even checksum). In such cases, we should delete and re-download the same file.